### ISSUE TITLE
[new feature] Make an option to save flags values as hash instead of plaintext

### ISSUE THREAD

**Issue by pun-private** [CTFd/CTFd#2176]
When adding a new challenge to CTFD, we have to specify the flag. For admin users, we can view the flag in the challenge info from the admin panel.

Is it possible to add an option to hash its value so :

- if someone manages to get access challenge details from the admin panel, only the hash is displayed
- when a user submits a flag, it is first hashed then compared to the hash value stored in the database.

Thank you for your time !

**Comment by ColdHeat** [comment]
I don't think this would be possible with how CTFd works. Especially if you consider things like case sensitivity and regex flags. 

If you propose a way that it could work I am open to the idea but it just sounds like it would be too difficult.

**Comment by pun-private** [comment]
I was thinking of creating a new type of flag. Today there are two types : static and dynamic.

What about creating a new type of flag like 'hashed' ?

**Comment by ColdHeat** [comment]
That wouldn't really achieve much because you would just store a hash value which you could have done using the static flag.

**Comment by GramThanos** [comment]
> That wouldn't really achieve much because you would just store a hash value which you could have done using the static flag.

The difference would be that the admins will not be able to see the flags.
